integration/spark/src/main/scala/org/apache/spark/sql/execution/strategy/CarbonLateDecodeStrategy.scala

```diff
@@ -89,7 +104,7 @@ private[sql] class CarbonLateDecodeStrategy extends SparkStrategy {
         try {
           pruneFilterProject(
             l,
-            projects.filterNot(_.name.equalsIgnoreCase(CarbonCommonConstants.POSITION_ID)),
```

Comment:
why is the filter removed?

Reply:
Because these position id should not be removed always, it has to be removed only in certain conditions (example, when `isPositionIDRequested` property is not set) 

Inside this `getRequestedColumns` will take care of removing it based on conditions.